content/spin/v1/dynamic-configuration.md

```diff
@@ -5,65 +5,56 @@ date = "2022-03-14T00:22:56Z"
-section. These entries aren't accessed directly by components but are referenced
-by [component config](#component-custom-config) value templates. Each entry must
-either have a `default` value or be marked as `required = true`. "Required" entries
-must be [provided](#custom-config-providers) with a value.
-
-Configuration keys may only contain lowercase letters and underscores between letters:
-
-<!-- @nocpy -->
-
-```toml
-[variables]
-api_host = { default = "api.example.com" }
-api_key = { required = true }
-```
-
-### Component Custom Config
-
-The configuration entries available to a component are listed in its
-`[component.config]` section. Configuration values may reference
-[config variables](#custom-config-variables) with simple
-[mustache](https://mustache.github.io/)-inspired string templates:
-
-<!-- @nocpy -->
-
-```toml
-[[component]]
-# ...
-[component.config]
-api_base_url = "https://{{ api_host }}/v1"
-api_key = "{{ api_key }}"
-```
-
-## Custom Config Providers
-
-[Custom config variables](#custom-config-variables) values may be set at runtime by
-config "providers". Currently, there are two providers: the environment
-variable provider and vault config provider.
+  - [Vault Application Variable Provider](#vault-application-variable-provider)
+    - [Vault Application Variable Provider Example](#vault-application-variable-provider-example)
+- [Key Value Store Runtime Configuration](#key-value-store-runtime-configuration)
+  - [Redis Key Value Store Provider](#redis-key-value-store-provider)
+  - [Azure CosmosDB Key Value Store Provider](#azure-cosmosdb-key-value-store-provider)
+- [SQLite Storage Runtime Configuration](#sqlite-storage-runtime-configuration)
+  - [LibSQL Storage Provider](#libsql-storage-provider)
+- [LLM Runtime Configuration](#llm-runtime-configuration)
+- [Remote Compute Provider](#remote-compute-provider)
+
+Configuration for Spin applications using features such as [application variables](https://developer.fermyon.com/spin/variable),
+[key value storage](https://developer.fermyon.com/spin/kv-store-api-guide), [SQL storage](https://developer.fermyon.com/spin/sqlite-api-guide)
+and [Serverless AI](https://developer.fermyon.com/spin/serverless-ai-api-guide) can be provided
+dynamically via [runtime configuration](#runtime-configuration).
+
+For many Spin features, configuration for the backing implementation can be supplied dynamically, i.e. during the application runtime,
+requiring no changes to the application code itself. We call these backing implementations 'providers' and their configuration data
+is stored in the runtime configuration file (`runtime-config.toml`). This file is then passed in via the `--runtime-config-file` flag
+when invoking the `spin up` command.
+
+The list of Spin features supporting runtime configuration includes:
+
+- [Application Variables Runtime Configuration](#application-variables-runtime-configuration)
```

Comment:
This is replicating the entire page TOC which feels like a maintenance woe.  @vdice originally had this just being a list of feature areas rather than a full TOC which I think is more maintainable and easier to navigate.